lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/DataCodec.java

```diff
@@ -64,30 +69,33 @@
         XML_FACTORY.setProperty(XMLOutputFactory.IS_REPAIRING_NAMESPACES, true);
     }
 
-    private final BindingToNormalizedNodeCodec codec;
+    private final AdapterContext codec;
     private final DeserializeIdentifierCodec deserializeIdentifierCodec;
     private final SerializeIdentifierCodec serializeIdentifierCodec;
     private final JsonNodeConverter jsonNodeConverter;
     private final XmlNodeConverter xmlNodeConverter;
-    private final SchemaContext schemaContext;
+    private final EffectiveModelContext effectiveModelContext;
 
-    public DataCodec(final SchemaContext schemaContext) {
-        this.schemaContext = requireNonNull(schemaContext);
-        final BindingNormalizedNodeCodecRegistry registry = new BindingNormalizedNodeCodecRegistry();
-        this.codec = new BindingToNormalizedNodeCodec(GeneratedClassLoadingStrategy.getTCCLClassLoadingStrategy(),
-            registry);
-        this.codec.onGlobalContextUpdated(schemaContext);
+    public DataCodec(final EffectiveModelContext effectiveModelContext) {
+        this.effectiveModelContext = requireNonNull(effectiveModelContext);
+        DefaultBindingRuntimeGenerator bindingRuntimeGenerator = new DefaultBindingRuntimeGenerator();
+        BindingRuntimeTypes runtimeTypes = bindingRuntimeGenerator.generateTypeMapping(effectiveModelContext);
+        BindingRuntimeContext bindingRuntimeContext = DefaultBindingRuntimeContext.create(runtimeTypes,
+                GeneratedClassLoadingStrategy.getTCCLClassLoadingStrategy());
+        BindingCodecContext bindingCodecContext = new BindingCodecContext(bindingRuntimeContext);
```

Comment:
Hmm, don't we already have this somewhere else (among Binding Runtime?)

Reply:
also ... since we are sourcing our YANG files via YangModuleInfo, we really want to talk in terms of BindingRuntimeContext, and not use GeneratedClassLoadingStrategy